packages/template-retail-react-app/app/components/recommended-products/index.jsx

```diff
@@ -164,7 +188,9 @@ const RecommendedProducts = ({zone, recommender, products, title, shouldFetch, .
                     )
                 },
                 enableFavourite: true,
-                isFavourite: !!wishlist.findItemByProductId(product?.productId),
+                isFavourite: !!wishlist?.customerProductListItems?.find(
+                    (item) => item.productId === product?.productId
+                ),
```

Comment:
```suggestion
                isFavourite: wishlist?.customerProductListItems?.some(
                    (item) => item.productId === product?.productId
                ),
```
If `boolean | undefined` is allowed for `isFavourite`, we can avoid the `!!`. (If `undefined` is not allowed, then `.find` vs `.some` doesn't really make a difference here.)